src/malli/transform.cljc

```diff
@@ -379,21 +364,27 @@
     :encoders (-string-encoders)}))
 
 (defn strip-extra-keys-transformer
-  ([]
-   (strip-extra-keys-transformer nil))
+  ([] (strip-extra-keys-transformer nil))
   ([{:keys [accept] :or {accept (m/-comp #(or (nil? %) (true? %)) :closed m/properties)}}]
-   (let [transform {:compile (fn [schema _]
-                               (when (accept schema)
-                                 (when-let [ks (some->> schema m/entries (map first) seq set)]
-                                   (fn [x]
-                                     ;; accept checks if the schema is compatible with strip-extra-keys,
-                                     ;; but the value might not be compatible with reduce-kv, i.e. a string.
-                                     (if (map? x)
-                                       (reduce-kv (fn [acc k _] (if-not (ks k) (dissoc acc k) acc)) x x)
-                                       x)))))}]
+   (let [strip-map {:compile (fn [schema _]
+                               (let [default-schema (m/default-schema schema)
+                                     ks (some->> schema (m/explicit-keys) (set))]
+                                 (cond-> nil
+                                   (accept schema)
+                                   (assoc :enter (fn [x]
+                                                   (if (and (map? x) (not default-schema))
```

Comment:
where's the part that strips the keys that don't pass default-schema? oh right, it's in the map `-transformer` method.

Reply:
yes!